### PR TITLE
PR 5: rename common module aliases, lint macros -> fragments, moog bundle support, node 14

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: lib/upgrader.js [hunk 1]
```diff
@@ -68,6 +125,33 @@ module.exports = ({ argv }) => {
     additional code upgrade concerns requiring manual attention.
   `);
 
+  function processLibModules() {
```
Comment (boutell) [inline lib/upgrader.js:136]: Refatored from the single site project case.

File: lib/linters.js [hunk 2]
```diff
@@ -144,6 +146,16 @@ const html = {
       of a piece type, page type or widget with "max" set to 1.
     `,
     url: 'https://v3.docs.apostrophecms.org/guide/upgrading.html#areas-and-pages'
+  },
+  macro: {
+    match: /{%\s*macro/g,
+    message: stripIndent`
+      In A3, Nunjucks macros cannot contain areas, but fragments can.
```
Comment (abea) [inline lib/linters.js:153]: For PRO-2317 this should warn about `with context`

File: lib/upgrader.js [hunk 3]
```diff
@@ -51,14 +103,26 @@ module.exports = ({ argv }) => {
...
+    if (!fs.existsSync('./lib/modules')) {
+      // TODO fully parse out the directory setting in index.js, but
+      // this is a strong norm
+      fail('Currently moog bundles are only supported if they have a lib/modules directory.');
+    }
+    processLibModules();
+    processStandaloneModule();
+    console.log(stripIndent`
+      ⚠️ lib/modules has been renamed to modules, but you will need to
+      manually change lib/modules to modules in the index.js file of
```
Comment (abea) [inline lib/upgrader.js]: Quotation marks around `lib/modules` and `modules` would be helpful here.

File: lib/upgrader.js [hunk 4]
```diff
@@ -68,6 +132,33 @@ module.exports = ({ argv }) => {
     additional code upgrade concerns requiring manual attention.
   `);
 
+  function processLibModules() {
+    // Only touch directories, don't mess about with regular files in lib/modules (WTF)
```
Comment (abea) [inline lib/upgrader.js]: What's the "WTF"? We should avoid vague reaction comments in favor of clear instructions.

File: lib/upgrader.js [hunk 5]
```diff
@@ -68,6 +132,33 @@ module.exports = ({ argv }) => {
...
+    // or symlinks in lib/modules (double WTF, but outside our remit)
```
Comment (abea) [inline lib/upgrader.js]: Same here with "double WTF"

File: lib/upgrader.js [hunk 6]
```diff
@@ -68,6 +132,33 @@ module.exports = ({ argv }) => {
...
+      rename(`lib/modules/${name}`, `modules/${name}`);
+    });
+    try {
+      fs.rmdirSync('lib/modules');
+    } catch (e) {
+      if (e.code === 'ENOTEMPTY') {
+        console.error(stripIndent`
+          "lib/modules" is not empty after moving modules to "modules". You probably
+          have files that are not apostrophe modules in that folder. Please
+          move those files to a more appropriate location, like lib, then
```
Comment (abea) [inline lib/upgrader.js]: quote "lib"

File: lib/upgrader.js [hunk 7]
```diff
@@ -68,6 +132,33 @@ module.exports = ({ argv }) => {
...
+    moduleNames.forEach(name => {
+      rename(`lib/modules/${name}`, `modules/${name}`);
+    });
+    try {
+      fs.rmdirSync('lib/modules');
+    } catch (e) {
+      if (e.code === 'ENOTEMPTY') {
+        console.error(stripIndent`
+          "lib/modules" is not empty after moving modules to "modules". You probably
+          have files that are not apostrophe modules in that folder. Please
```
Comment (abea) [inline lib/upgrader.js:149]: Capitalize "apostrophe"

File: lib/upgrader.js [hunk 8]
```diff
@@ -23,6 +26,55 @@ module.exports = ({ argv }) => {
...
+    [ 'apos.templates', 'apos.template' ],
+    [ 'apos.videoFields', 'apos.videoField' ],
+    [ 'apos.pages', 'apos.page' ],
+    // No blanket replace for areas because we don't want to confuse the
+    // separate issue of linting the "apos.area" helper by renaming "apos.areas"
+    // to that. These are the individual likely offenders
+    [ 'apos.areas.getWidgetManager', 'apos.area.getWidgetManager' ],
+    [ 'apos.areas.richText', 'apos.area.richText' ],
+    [ 'apos.areas.plaintext', 'apos.area.plaintext' ],
+    [ 'apos.areas.walk', 'apos.area.walk' ]
```
Comment (abea) [inline lib/upgrader.js:62]: `apos.areas.isEmpty` should be included here.

Comment (abea) [reply lib/upgrader.js:62]: It looks like `apos.area.richText` still triggers the "Use the new area syntax" linter message. Clemens.

**Review by abea — CHANGES_REQUESTED** [review]
It looks like there might be another eslint err.

- A converted bundle module (I used apos-blog) leaves the `directory: 'lib/modules'` setting despite moving that directory.

**Comment by abea** [comment]
It's worth noting that this is a pretty badass tool. So many things to cover and catch. This is really nice work.

**Comment by boutell** [comment]
"A converted bundle module (I used apos-blog) leaves the directory: 'lib/modules' setting despite moving that directory."

This is something we warn the developer about (for now).

**Review by abea — APPROVED** [review]

**Pull request merged**